src/MCPClient/lib/clientScripts/parse_external_mets.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python2
+
+from __future__ import print_function
+import argparse
+from lxml import etree
+import os
+import sys
+
+# archivematicaCommon
+from custom_handlers import get_script_logger
+
+from archivematicaAssignFileUUID import find_mets_file
+import parse_mets_to_db
+
+logger = get_script_logger('archivematica.mcp.client.parse_external_mets')
+
+def parse_reingest_mets(transfer_uuid, transfer_path):
+    # Parse METS to extract information needed by later microservices
+    mets_path = find_mets_file(transfer_path)
+    try:
+        root = etree.parse(mets_path)
+    except Exception:
+        print('Error parsing reingest METS', mets_path, ' - skipping')
+        logger.info('Error parsing reingest mets %s - skipping', mets_path, exc_info=True)
+        return
+
+    # Get SIP UUID from METS name
+    sip_uuid = os.path.basename(mets_path).replace('METS.', '').replace('.xml', '')
+    # TODO Create stub SIP for the metadata to be associated with
+    dc = parse_mets_to_db.parse_dc(sip_uuid, root)
+    rights_list = parse_mets_to_db.parse_rights(sip_uuid, root)
+
+    if not dc and not rights_list:
+        # TODO Delete stub SIP
+        pass
```

Comment:
Ah, this was a note to handle the case on reingest where it's a new pipeline and the SIP object doesn't exist anymore. Since the DC & Rights are associated with the SIP, I thought this would cause a foreign key error. Since it's only semantically a foreign key, but not enforced in the DB (because it can go to SIP or Transfer table), it's fine. I left a note in the code to that effect.